q2_fmt/_peds.py

```diff
@@ -8,30 +8,191 @@
+        })
+        peds_df['measure'].attrs.update({
+            'title': "PEDS",
+            'description': 'Proportional Engraftment of Donor Strains '
+        })
+        peds_df['group'].attrs.update({
+            'title': time_column,
+            'description': 'Time'
+        })
+        peds_df["subject"].attrs.update({
+            'title': subject_column,
+            'description': 'ID to link samples across time'
+        })
+        peds_df["transfered_donor_features"].attrs.update({
+            'title': "Transfered Donor Features",
+            'description': '...'
+        })
+        peds_df['total_donor_features'].attrs.update({
+            'title': "Total Donor Features",
+            'description': '...'
+        })
+        peds_df['donor'].attrs.update({
+            'title': reference_column,
+            'description': 'Donor'
+        })
+
+    elif peds_type == "Feature":
+        num_sum = np.sum(maskedrecip, axis=0)
+        donor_sum = np.sum(donormask, axis=0)
+        for count, feature in enumerate(recipdf.columns):
+            with warnings.catch_warnings():
+                warnings.simplefilter("ignore")
+                peds = num_sum[count] / donor_sum[count]
+            peds_df.loc[len(peds_df)] = [feature, peds, num_sum[count],
+                                         donor_sum[count], peds_time, feature]
+            peds_df = peds_df.dropna()
+        peds_df['id'].attrs.update({
+            'title': "FeatureID",
+            'description': ''
+        })
+        peds_df['measure'].attrs.update({
+            'title': "PEDS",
+            'description': 'Proportional Engraftment of Donor Strains '
+        })
+        peds_df['group'].attrs.update({
+            'title': time_column,
+            'description': 'Time'
+        })
+        peds_df['subject'].attrs.update({
+            'title': "FeatureID",
+            'description': ''
+        })
+    else:
+        raise KeyError('There was an error finding which PEDS methods to use')
+    return peds_df
+
+
+# Filtering methods
+def _check_for_time_column(metadata, time_column):
     try:
         num_timepoints = metadata[time_column].dropna().unique().size
     except Exception as e:
         raise KeyError('There was an error finding %s in the metadata'
                        % time_column) from e
+    return num_timepoints
+
+
+def _check_time_column_numeric(metadata, time_column):
     try:
         metadata[time_column].dropna().astype(int)
     except Exception as e:
         raise ValueError('%s must be numeric' % time_column) from e
 
+
+def _check_reference_column(metadata, reference_column):
     try:
         reference_series = metadata[reference_column]
     except Exception as e:
         raise KeyError('There was an error finding %s in the metadata'
                        % reference_column) from e
+    return reference_series
+
+
+def _check_associated_reference(reference_series, metadata, time_column,
```

Comment:
```suggestion
def _filter_associated_references(reference_series, metadata, time_column,
```
Also can take it or leave it on this suggestion, but it might be a bit more clear to call this one a `filter` helper instead of a `check` helper - since you aren't checking for something specific here, just pulling out a subset of `reference_series` and returning that with the associated metadata.

Reply:
Changed this outside of github so I could address when it was called too.